src/utils.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace LanguageServer;
+
+/**
+ * Recursively Searches files with matching filename, starting at $path. 
+ *
+ * @param string $path
+ * @param string $pattern
+ * @return array
+ */
+function findFilesRecursive(string $path, string $pattern): array {
+    $dir = new \RecursiveDirectoryIterator($path);
+    $ite = new \RecursiveIteratorIterator($dir);
+    $files = new \RegexIterator($ite, $pattern, \RegexIterator::GET_MATCH);
+    $fileList = array();
+    foreach($files as $file) {
+        $fileList = array_merge($fileList, $file);
+    }
+    return $fileList;
+}
+
+/**
+ * Transforms an absolute file path into a URI as used by the language server protocol. 
+ *
+ * @param string $filepath
+ * @return string
+ */
+function pathToUri(string $filepath): string {
+    return 'file://'.($filepath[0] == '/' || $filepath[0] == '\\' ? '' : '/').str_replace('\\', '/', $filepath);

```

Comment:
I think this implementation is not correct. A file URI must _always_ start with `file:///` (triple slash). In your case, you are checking if the first char is a backslash, but on windows it could be a drive letter. Also, special chars need to be URL-encoded. There is a very good implementation I've been using on NPM: https://github.com/sindresorhus/file-url/blob/master/index.js#L17-L24